File: 242.valid-anagram.py
#
# @lc app=leetcode id=242 lang=python
#
# [242] Valid Anagram
#

# @lc code=start
from collections import defaultdict


class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        his = defaultdict(int)
        if len(s) != len(t):
            return False
        for i in s:
            his[i] += 1
        for i in t:
            his[i] -= 1
        for i in his.values():
            if i != 0:
                return False
        return True
# @lc code=end

File: 242.valid-anagram.java
/*
 * @lc app=leetcode id=242 lang=java
 *
 * [242] Valid Anagram
 */

// @lc code=start
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] his = new int[26];
        if (s.length() != t.length()) {
          return false;
        }
        for(int i = 0; i<s.length(); i++) {
            int sc = s.charAt(i) - 'a';
            int tc = t.charAt(i) - 'a';
            his[sc]++;
            his[tc]--;
        }
        for (int i : his) {
            if (i != 0) {
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

